basket-analysis/basket-analysis-quickstart/Basket_Analysis.ipynb

```diff
@@ -0,0 +1,709 @@
+    "\tL_PARTKEY NUMBER(38,0),\n",
+    "\tL_SUPPKEY NUMBER(38,0),\n",
+    "\tL_LINENUMBER NUMBER(38,0),\n",
+    "\tL_QUANTITY NUMBER(12,2),\n",
+    "\tL_EXTENDEDPRICE NUMBER(12,2),\n",
+    "\tL_DISCOUNT NUMBER(12,2),\n",
+    "\tL_TAX NUMBER(12,2),\n",
+    "\tL_RETURNFLAG VARCHAR(1),\n",
+    "\tL_LINESTATUS VARCHAR(1),\n",
+    "\tL_SHIPDATE DATE,\n",
+    "\tL_COMMITDATE DATE,\n",
+    "\tL_RECEIPTDATE DATE,\n",
+    "\tL_SHIPINSTRUCT VARCHAR(44),\n",
+    "\tL_SHIPMODE VARCHAR(10),\n",
+    "\tL_COMMENT VARCHAR(44)\n",
+    ");"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6618a464-fc88-4e5e-8a6b-4a8fc0c5cb81",
+   "metadata": {
+    "language": "sql",
+    "name": "cell10"
+   },
+   "outputs": [],
+   "source": [
+    "\n",
+    "CREATE OR REPLACE TABLE TPCH_EXAMPLE.GDS.PARTS(\n",
+    "\tP_PARTKEY NUMBER(38,0),\n",
+    "\tP_NAME VARCHAR(55),\n",
+    "\tP_MFGR VARCHAR(25),\n",
+    "\tP_BRAND VARCHAR(10),\n",
+    "\tP_TYPE VARCHAR(25),\n",
+    "\tP_SIZE NUMBER(38,0),\n",
+    "\tP_CONTAINER VARCHAR(10),\n",
+    "\tP_RETAILPRICE NUMBER(12,2),\n",
+    "\tP_COMMENT VARCHAR(23)\n",
+    ");"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f10ecf03-fbef-4854-9fee-887a45487f1c",
+   "metadata": {
+    "collapsed": false,
+    "name": "cell11"
+   },
+   "source": [
+    "[Stages](https://docs.snowflake.com/en/user-guide/data-load-local-file-system-create-stage.html) in snowflake are places that you can land your data before it is uploaded to a Snowflake table. You might have a batch of CSV files living on a disk driver somewhere, and, in order to start querying the data via a table, the data must be landed within the Snowflake environment for a data upload to be possible.\n",
+    "\n",
+    "In the exercise, we will be working with structured, comma-delimited data that has already been staged in a public, external AWS bucket. Before we can use this data, we first need to create a `Stage` that specifies the location of our external bucket.\n",
+    "\n",
+    "Let’s create the \"stage\" object. \"Stages\" are typically created by Storage Administrators, but for the purposes of this lab, YOU will be creating this object. Again, lets click back on \"Worksheets\" tab  on the left hand side and excute the next following lines of code:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b2e9bbe2-cf86-4db5-bbc1-e18cfe2fb2db",
+   "metadata": {
+    "language": "sql",
+    "name": "cell12"
+   },
+   "outputs": [],
+   "source": [
+    "-- you must be accountadmin to create role and grant permissions\n",
+    "USE ROLE accountadmin;"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ca7b2c44-57db-49ed-82a1-d6bd3078dec3",
+   "metadata": {
+    "language": "sql",
+    "name": "cell13"
+   },
+   "outputs": [],
+   "source": [
+    "CREATE OR REPLACE STAGE sf_neo4j_data_stage\n",
+    "  URL = 's3://neo4j-snowflake-data/tpc-h/';"
```

Comment:
Is it the same data as here?
https://docs.snowflake.com/en/user-guide/sample-data-tpch

If yes, I think its better to use it from sf. In this data can be easily obtain with these queries without having a stage
```
-- Create a database from the share.
CREATE DATABASE SNOWFLAKE_SAMPLE_DATA FROM SHARE SFC_SAMPLES.SAMPLE_DATA;

-- Grant the PUBLIC role access to the database.
-- Optionally change the role name to restrict access to a subset of users.
GRANT IMPORTED PRIVILEGES ON DATABASE SNOWFLAKE_SAMPLE_DATA TO ROLE PUBLIC;
```

see https://docs.snowflake.com/en/user-guide/sample-data-using